pkg/spec/storage.go

```diff
@@ -856,45 +856,22 @@ func SupercedeUserMounts(mounts []spec.Mount, configMount []spec.Mount) []spec.M
 
 // Ensure mount options on all mounts are correct
 func InitFSMounts(inputMounts []spec.Mount) ([]spec.Mount, error) {
-	// We need to look up mounts so we can figure out the proper mount flags
-	// to apply.
-	systemMounts, err := pmount.GetMounts()
-	if err != nil {
-		return nil, errors.Wrapf(err, "error retrieving system mounts to look up mount options")
-	}
-
 	// TODO: We probably don't need to re-build the mounts array
-	var mounts []spec.Mount
+	mounts := make([]spec.Mount, len(inputMounts))
```

Comment:
```suggestion
	mounts := make([]spec.Mount, 0, len(inputMounts))
```
Shouldn't this initialize capacity to `len(inputMounts)`, but keep the index at 0, since we're appending?

Reply:
You're right! But the more I look into users of this function, the more I think we should just fix things in place.

Let me fix it.

Reply:
OK I have added a separate commit which changes the function to fix options in place, and convert its users.